File: app/src/main/java/se/creotec/chscardbalance2/model/IModel.java
// Copyright (c) 2017 Alexander Håkansson
//
// This software is released under the MIT License.
// https://opensource.org/licenses/MIT
package se.creotec.chscardbalance2.model;

import android.support.annotation.NonNull;

public interface IModel {
    void setCardData(final CardData data);
    CardData getCardData();
    void addCardDataListener(@NonNull OnCardDataChangedListener listener);
    void notifyCardDataChangedListeners();

    void setMenuData(final MenuData data);
    MenuData getMenuData();
    void addMenuDataListener(@NonNull OnMenuDataChangedListener listener);
    void notifyMenuDataChangedListeners();

}

File: app/src/main/java/se/creotec/chscardbalance2/model/IModel.kt
// Copyright (c) 2017 Alexander Håkansson
//
// This software is released under the MIT License.
// https://opensource.org/licenses/MIT
package se.creotec.chscardbalance2.model

import se.creotec.chscardbalance2.service.AbstractBackendService
import kotlin.reflect.KClass

interface IModel {
    var cardData: CardData
    fun addCardDataListener(listener: OnCardDataChangedListener)
    fun notifyCardDataChangedListeners()
    var cardLastTimeUpdated: Long

    var menuData: MenuData
    fun addMenuDataListener(listener: OnMenuDataChangedListener)
    fun notifyMenuDataChangedListeners()
    var preferredMenuLanguage: String
    var menuLastTimeUpdated: Long

    val quickChargeURL: String

    fun addServiceFailedListener(listener: OnServiceFailedListener)
    fun notifyServiceFailed(service: AbstractBackendService<*>, error: String)

    interface OnServiceFailedListener {
        fun serviceFailed(service: AbstractBackendService<*>, error: String)
    }

}
